cpp/dolfinx/function/interpolate.h

```diff
@@ -237,6 +182,40 @@ void interpolate(
   const int value_size = std::accumulate(std::begin(vshape), std::end(vshape),
                                          1, std::multiplies<>());
 
+  auto mesh = u.function_space()->mesh();
+  assert(mesh);
+  const int tdim = mesh->topology().dim();
+  if (element->family() == "Mixed")
+  {
+    // Extract the correct
+    auto dofmap = u.function_space()->dofmap();
+    auto map = mesh->topology().index_map(tdim);
+    assert(map);
+    const int num_cells = map->size_local() + map->num_ghosts();
+
+    Eigen::Array<T, 1, Eigen::Dynamic> mixed_values(values.cols());
+    int value_offset = 0;
+    for (int i = 0; i < element->num_sub_elements(); ++i)
+    {
+      const std::vector<int> component = {i};
+      auto sub_dofmap = dofmap->extract_sub_dofmap(component);
```

Comment:
I find it more readable to me explicit with the type here rather than using `auto`.